docs/docs/09-commands-and-tools/02-ide-support.md

```diff
@@ -37,80 +37,156 @@ Include the following to the settings.json in order to enable autocompletion for
+```lua
+vim.filetype.add({ extension = { templ = "templ" } })
+```
 
+##### Other LSPs within .templ files
+
+These LSPs can be used *in conjunction* with the templ lsp and tree-sitter. Here's how to set them up.
+
+[html-lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#html) - First make sure you have it installed `:LspInstall html` or find it on the `:Mason` list. 
 ```lua
--- additional filetypes
-vim.filetype.add({
- extension = {
-  templ = "templ",
- },
+lspconfig.html.setup({
+    on_attach = on_attach,
+    capabilities = capabilities,
+    filetypes = { "html", "templ" },
 })
 ```
 
-### Format on Save
+[htmx-lsp](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#htmx) - First make sure you have it installed `:LspInstall htmx` or find it on the `:Mason` list. Note with this LSP, it activates after you type `hx-` in an html attribute, because that's how all htmx attributes are written.
+```lua
+lspconfig.htmx.setup({
+    on_attach = on_attach,
+    capabilities = capabilities,
+    filetypes = { "html", "templ" },
+})
+```
+
+[tailwindcss](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tailwindcss) - First make sure you have it installed `:LspInstall tailwindcss` or find it on the `:Mason` list.
+```lua
+lspconfig.tailwindcss.setup({
+    on_attach = on_attach,
+    capabilities = capabilities,
+    filetypes = { "templ", "astro", "javascript", "typescript", "react" },
+    init_options = { userLanguages = { templ = "html" } },
+})
+```
+
+Inside of your `tailwind.config.js`, you need to tell tailwind to look inside of .templ files and/or .go files.
+```js
+module.exports = {
+    content: [ "./**/*.html", "./**/*.templ", "./**/*.go", ],
+    theme: { extend: {}, },
+    plugins: [],
+}
+```
+
+### Formatting
 
 With the templ LSP installed and configured, you can use the following code snippet to format on save:
 
 
 ```lua
--- Format current buffer using LSP.
-vim.api.nvim_create_autocmd(
-  {
-    -- 'BufWritePre' event triggers just before a buffer is written to file.
-    "BufWritePre"
-  },
-  {
-    pattern = {"*.templ"},
-    callback = function()
-      -- Format the current buffer using Neovim's built-in LSP (Language Server Protocol).
-      vim.lsp.buf.format()
-    end,
-  }
-)
+vim.api.nvim_create_autocmd({ "BufWritePre" }, { pattern = { "*.templ" }, callback = vim.lsp.buf.format })
 ```
+`BufWritePre` means that the callback gets ran after you call `:write`.
 
-### Tailwind CSS Intellisense
+If you have multiple LSPs attached to the same buffer, and you have issues with `vim.lsp.buf.format`, you can use this snippet to run `templ fmt` in the same way that you might from the command line.
+This will get the buffer and its corresponding filename, and refresh the buffer after it has been formatted so you don't get out of sync issues.
```

Comment:
```suggestion

This will get the buffer and its corresponding filename, and refresh the buffer after it has been formatted so you don't get out of sync issues.
```